wdl-gauntlet/src/lib.rs

```diff
@@ -241,14 +241,14 @@ pub async fn gauntlet(args: Args) -> Result<()> {
             // As the list of diagnostics has been sorted by document identifier, do
             // a binary search and collect the matching messages
             let diagnostics = config.inner().diagnostics();
-            let doc_id_str = document_identifier.to_string();
+            let doc_id = document_identifier.to_string();
             let expected: IndexSet<String> = diagnostics
-                .binary_search_by_key(&doc_id_str.as_str(), |d| d.document())
+                .binary_search_by_key(&doc_id, |d| d.document().to_string())
```

Comment:
Does `doc_id` need to be a string now that the field is an `Identifier`?

Could we not just pass `&document_identifier` and do `d.document()` in the closure?

Reply:
Is there a relevant difference between doing `*d.document() == document_identifier` and `d.document() == &document_identifier`?

Regardless, good call that makes this section easier to read.

Reply:
It makes no different between the two.